### PR TITLE
Change README to Markdown format, add LICENSE.txt

### PR DESCRIPTION
**Pull request by rubenwardy**
[See rendered version](https://github.com/rubenwardy/minetest/blob/readmemd/README.md)

### CODE REVIEW

**Comment by bigfoot547** [comment]
WE NEED MORE SHIELDS! :smile:

**Comment by sfan5** [comment]
shouldn't LICENSE.txt also get installed?

**Comment by rubenwardy** [comment]
> shouldn't LICENSE.txt also get installed?

done

> WE NEED MORE SHIELDS! :smile:

:D
 any suggestions for useful ones?

File: README.md [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+- You can select between Release and Debug build by -DCMAKE_BUILD_TYPE=<Debug or Release>
+  - Debug build is slower, but gives much more useful output in a debugger
+- If you build a bare server, you don't need to have Irrlicht installed.
+  In that case use -DIRRLICHT_SOURCE_DIR=/the/irrlicht/source
+
+### CMake options
+
+General options:
+
+* BUILD_CLIENT        - Build Minetest client
```
Comment (SmallJoker) [inline README.md]: I don't like how the indents are lost in the Markdown projection. How about putting these params into a code block or table?

Comment (paramat) [inline README.md]: I agree this section is hard on the eyes now.

**Comment by nerzhul** [comment]
another PR ? please close redundant PR then

**Comment by rubenwardy** [comment]
> another PR ? please close redundant PR then

Already did that when I opened this PR

**Pull request merged**